prov/shm/src/smr_signal.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019 Amazon.com, Inc. or its affiliates.
+ * All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _SMR_SIGNAL_H_
+#define _SMR_SIGNAL_H_
+#include <signal.h>
+#include <dirent.h>
+#include <ofi_shm.h>
+
+struct sigaction old_action;
+
+static void smr_handle_signal(int signum, siginfo_t *info, void *ucontext)
+{
+	int ret;
+
+	shm_unlink(smr_ep_name);
+
+	/* Register the original signum handler, SIG_DFL or otherwise */
+	ret = sigaction(signum, &old_action, NULL);
+	if (ret)
+		return;
+
+	/* Raise signum to execute the original handler */
+	ret = raise(signum);
+	if (ret)
+		return;
```

Comment:
The if statement doesn't do anything here.  Maybe add a warning or just drop.

Reply:
Unsafe to add a print within a handler, but I will drop it.